### DISCUSSION TITLE
Include cells with 2+ guides in low MOI analysis

### DISCUSSION THREAD

**Discussion by argschwind** [Katsevich-Lab/sceptre#140] [Q&A] [answered]
Hi,

We use Sceptre to analyze an enhancer targeting TAP-seq screen from our lab. It's a low MOI experiment, so naturally we use the low MOI setting in Sceptre. Even at low MOI there's always a small fraction of cells with 2+ guides, but we have noticed that with the low MOI setting these cells are rigorously filtered out.

Is there an easy way you could recommend to include cells with 2+ guides in the discovery analyses?

We also tried running Sceptre in the high MOI setting but observed model calibration issues.

Thanks in advance for you help!

**Comment by timothy-barry** [comment]
Hi Andreas,

Good to hear from you. I guess this an issue related to the control group. Would the idea be to use cells that contain a (single) NT gRNA as the control group, while keeping cells that contain 2+ targeting gRNAs?

**Reply by argschwind** [reply]
Yes, exactly that's what we were thinking. Use only cells with NT guides as controls, but allow 2+ gRNAs for cells with enhancer targeting guides.

**Reply by ekatsevi** [reply]
@timothy-barry: Would it be possible to accomplish this by setting `moi = "high"` during import and then setting `control_group = "nt_cells"` in `set_analysis_parameters()`?

**Reply by timothy-barry** [reply]
Got it. No, it is not possible to set `moi` to `"high"` and `control_group` to `"nt_cells"`. Unfortunately, the functionality that Andreas is describing is not built into the package. I am thinking out loud here about how we might accomplish this. We could set `moi` to `"low"` and then remove cells that contain one (or more) NT gRNAs *in addition to* one (or more) targeting gRNAs. All other cells would be retained. I think this probably would work but I'd need to do some testing and experimentation. Anyone have any thoughts?

**Reply by argschwind** [reply]
I think for our purpose if a cell has an NT gRNA + a targeting (or more) gRNA it should still be used in the targeting group.

**Reply by ekatsevi** [reply]
I think this makes sense.

**Reply by timothy-barry** [reply]
I see. OK, I will allocate an hour on Wednesday to tinkering around with this. I will let you know what I come up with.

**Reply by argschwind** [reply]
Thank you so much for looking into this, Tim!

**Reply by argschwind** [reply]
Hi, are there any updates on this or could you let us know when we can expect some? I understand that you're probably busy with many other important things as well.

**Reply by timothy-barry** [reply]
Hey Andreas, working on this now. Nothing to share yet.

**Reply by timothy-barry** [reply]
I put some time into this today, and unfortunately, I think it will be challenging to implement this update to the package. I am going to give this another shot on Friday, but sadly I can't make any promises that this functionality will be available in the near future.

Do you think you might be able to take a crack at implementing this functionality? If so, I can point you to the relevant parts of the code over a Zoom call.

**Comment by timothy-barry** [comment]
Also -- If you observe good calibration out to 1e-3 or so, it should be reasonable to assume that the false discovery rate is controlled (provided that there are some true signals among your discovery pairs).

**Reply by argschwind** [reply]
We observe severe miscalibration with the high moi setting compared to low MOI. We use the same covariates in both analyses, so not sure what's happening here.

Low MOI:
![calibration_check](https://github.com/user-attachments/assets/d0fe8bd2-b143-412a-95b9-651022a1c186)

High MOI:
![calibration_check_highmoi](https://github.com/user-attachments/assets/672627d0-f5a6-4402-9c0f-41e4deccb8aa)

**Reply by timothy-barry** [reply]
I see. Yes, the high MOI will need to be looked into.

**Comment by ekatsevi** [comment, marked as answer]
sceptre does not currently provide a supported way to include cells with 2+ guides in a low-MOI analysis. We explored this in [PR #175](https://github.com/Katsevich-Lab/sceptre/pull/175) and preserved the implementation on the [`archive/pr-175-flexible-cell-assignments`](https://github.com/Katsevich-Lab/sceptre/tree/archive/pr-175-flexible-cell-assignments) branch as experimental work, but it would require correctness fixes and renewed review before adoption.
